wpimath/src/main/java/edu/wpi/first/math/geometry/Ellipse2d.java

```diff
@@ -0,0 +1,192 @@
+  public Pose2d getCenter() {
+    return m_center;
+  }
+
+  /**
+   * Returns the x semi-axis.
+   *
+   * @return The x semi-axis.
+   */
+  public double getXSemiAxis() {
+    return m_xSemiAxis;
+  }
+
+  /**
+   * Returns the y semi-axis.
+   *
+   * @return The y semi-axis.
+   */
+  public double getYSemiAxis() {
+    return m_ySemiAxis;
+  }
+
+  /**
+   * Returns either of the focal points of the ellipse. In a perfect circle, this will always return
+   * the center.
+   *
+   * @param first Whether to return the first (-c) or second (+c) focal point.
+   * @return A focal point.
+   */
+  public Translation2d getFocalPoint(boolean first) {
+    double a = Math.max(m_xSemiAxis, m_ySemiAxis); // Major semi-axis
+    double b = Math.min(m_xSemiAxis, m_ySemiAxis); // Minor semi-axis
+    double c = Math.sqrt(a * a - b * b);
+
+    c = first ? -c : c;
+
+    if (m_xSemiAxis > m_ySemiAxis) {
+      Transform2d diff = new Transform2d(c, 0.0, new Rotation2d());
+      return m_center.plus(diff).getTranslation();
+    } else {
+      Transform2d diff = new Transform2d(0.0, c, new Rotation2d());
+      return m_center.plus(diff).getTranslation();
+    }
+  }
+
+  /**
+   * Transforms the center of the ellipse and returns the new ellipse.
+   *
+   * @param other The transform to transform by.
+   * @return The transformed ellipse.
+   */
+  public Ellipse2d transformBy(Transform2d other) {
+    return new Ellipse2d(m_center.transformBy(other), m_xSemiAxis, m_ySemiAxis);
+  }
+
+  /**
+   * Rotates the center of the ellipse and returns the new ellipse.
+   *
+   * @param other The rotation to transform by.
+   * @return The rotated ellipse.
+   */
+  public Ellipse2d rotateBy(Rotation2d other) {
+    return new Ellipse2d(m_center.rotateBy(other), m_xSemiAxis, m_ySemiAxis);
+  }
+
+  /**
+   * Solves the equation of an ellipse from the given point. This is a helper function used to
+   * determine if that point lies inside of or on an ellipse.
+   *
+   * <pre>
+   * (x-h)^2 / a^2 + (y-k)^2 / b^2 = 1
+   * </pre>
+   *
+   * @param point The point to solve for.
+   * @return < 1.0 if the point lies inside the ellipse, == 1.0 if a point lies on the ellipse, and
+   *     > 1.0 if the point lies outsides the ellipse.
+   */
+  private double solveEllipseEquation(Translation2d point) {
+    // Rotate the point by the inverse of the ellipse's rotation
+    point = point.rotateAround(m_center.getTranslation(), m_center.getRotation().unaryMinus());
+
+    double x = point.getX() - m_center.getX();
+    double y = point.getY() - m_center.getY();
+
+    return (x * x) / (m_xSemiAxis * m_xSemiAxis) + (y * y) / (m_ySemiAxis * m_ySemiAxis);
+  }
+
+  /**
+   * Checks if a point is intersected by this ellipse's circumference.
+   *
+   * @param point The point to check.
+   * @return True, if this ellipse's circumference intersects the point.
+   */
+  public boolean intersectsPoint(Translation2d point) {
+    return solveEllipseEquation(point) == 1.0;
```

Comment:
We might want to add a small tolerance (Usually we use 1e-9) to this, since usually double/float rounding errors will lead to values not being exactly equal to the desired result.

Reply:
That makes sense, I just added that.